sys/tracefd/include/tracefd.h

```diff
@@ -122,12 +151,24 @@ void tracefd_unlock(const tracefd_t tracefd);
 
 /**
  * @brief Print formatted string in the trace file
+ *        File descriptor must be locked/unlocked manually before/after
+ *        calling this function.
  * @param[in]   tracefd  trace file descriptor
  * @param[in]   format   format string
  * @param[in]   ...      arguments
  */
 void tracefd_printf(const tracefd_t tracefd, const char *format, ...);
 
+/**
+ * @brief Print formatted string in the trace file in a JSON log message.
+ *        The message must not contain '\n' nor '"'.
```

Comment:
Can these characters be escaped ?
Is ' allowed ?
If yes a tracefd_strip_jlog to replace these characters and send the json could be useful ? (in the future maybe)

Reply:
Forbidden characters could probably be escaped, but with a double-escape.
' is allowed.
I'm not in favor of automatically trying to escape unsupported characters because it would change the string length,
be time consuming, and not error prone.
So I would advise to pay attention to the strings given in argument of this function.
In any case, the JSON will be wrong but the program will not crash.